applications/GeoMechanicsApplication/custom_retention/van_genuchten_law.cpp

```diff
@@ -63,18 +60,18 @@ double VanGenuchtenLaw::CalculateEffectiveSaturation(Parameters& rParameters) co
 double VanGenuchtenLaw::CalculateDerivativeOfSaturation(Parameters& rParameters) const
 {
     KRATOS_TRY
-    const double& p = rParameters.GetFluidPressure();
+    const auto p = rParameters.GetFluidPressure();
 
     if (p > 0.0) {
-        const auto&   rMaterialProperties = rParameters.GetMaterialProperties();
-        const double& satMax              = rMaterialProperties[SATURATED_SATURATION];
-        const double& satMin              = rMaterialProperties[RESIDUAL_SATURATION];
-        const double& pb                  = rMaterialProperties[VAN_GENUCHTEN_AIR_ENTRY_PRESSURE];
-        const double& gn                  = rMaterialProperties[VAN_GENUCHTEN_GN];
-        const double  gc                  = (1.0 - gn) / gn;
-
-        return (satMax - satMin) * gc * pow((1.0 + pow(p / pb, gn)), gc - 1.0) * gn * pow(pb, -gn) *
-               pow(p, gn - 1.0);
+        const auto& rMaterialProperties = rParameters.GetMaterialProperties();
+        const auto  sat_max             = rMaterialProperties[SATURATED_SATURATION];
+        const auto  sat_min             = rMaterialProperties[RESIDUAL_SATURATION];
+        const auto  pb                  = rMaterialProperties[VAN_GENUCHTEN_AIR_ENTRY_PRESSURE];
+        const auto  gn                  = rMaterialProperties[VAN_GENUCHTEN_GN];
+        const auto  gc                  = (1.0 - gn) / gn;
```

Comment:
this is a copy of lines 33-37 above. Would it be better to extract them as a function to reduce duplication?